packages/video_player/ios/Classes/VideoPlayerPlugin.m

```diff
@@ -115,21 +191,46 @@ - (instancetype)initWithURL:(NSURL*)url frameUpdater:(FLTFrameUpdater*)frameUpda
           if (_disposed) return;
           if ([videoTrack statusOfValueForKey:@"preferredTransform" error:nil] ==
               AVKeyValueStatusLoaded) {
+            CGSize size = videoTrack.naturalSize;
+            NSLog(@"preferredTransform width, height: %f, %f", size.width, size.height);
+
+            // Rotate the video by using a videoComposition and the preferredTransform
+            _preferredTransform = [self fixTransform:videoTrack];
+            // Note:
+            // https://developer.apple.com/documentation/avfoundation/avplayeritem/1388818-videocomposition
+            // Aideo composition can only be used with file-based media and is not supported for
+            // use with media served using HTTP Live Streaming.
+            AVMutableVideoComposition* videoComposition =
+                [self getVideoCompositionWithTransform:_preferredTransform
+                                             withAsset:asset
+                                        withVideoTrack:videoTrack];
+            item.videoComposition = videoComposition;
             dispatch_async(dispatch_get_main_queue(), ^{
-              [_player replaceCurrentItemWithPlayerItem:item];
-            });
+                               // TODO explain why this line was here in the first place
+                               // The plugin seems to work OK without it
+                               // Even with this line, videos are sometimes blank in app
+                               // [_player replaceCurrentItemWithPlayerItem:item];
```

Comment:
I think if we get rid of this we should get rid of the `dispatch_async` block entirely, no?

I would lean towards keeping it though.  Unfortunately we don't seem to have a test for it, and the history where it was added is when it was first added to the repo.  If @sigurdm or @cbracken know better, maybe we can get rid of it - but in the absence of knowing what it might break I'd lean towards keeping it.

Reply:
I'll keep the block given that we are not sure what this is for.
I've updated the copy too with (hopefully) better wording.

Reply:
Thanks @sigurdm! If the docs have changed, should we then remove the async block?

Reply:
I am not longer maintainer of this project, but I see no reason to keep it.

Reply:
Got it. I'll remove the entire block. Thanks for your help!